src/ApiCompat/Rules/Compat/DifferenceRule.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System.Composition;
+
+namespace Microsoft.Cci.Differs.Rules
+{
+    internal abstract class DifferenceRule : Microsoft.Cci.Differs.DifferenceRule
+    {
+        [Import]
+        public IDifferenceOperands Operands { get; set; }
+
+        public string Left => Operands?.Left ?? "contract";
```

Comment:
I wonder if calling this Contract and Implementation may be easier to understand given all the rules are still written in that method? Or do you plan to do a pace to rename all the parameters as well :)

Reply:
Yeah I was thinking the same.  I think left and right is better if we were starting from scratch but maybe I should just stick to contract/implementation for now.